### PR TITLE
feat(tapable): upgrade other template plugins

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Refactoring

**Did you add tests for your changes?**
N/A

**If relevant, link to documentation update:**
N/A

**Summary**
This PR upgrades all of the MainTemplatePlugins to use the Tapable v1 Plugin API (using `.tap` instead of `.plugin`)

**Does this PR introduce a breaking change?**
Nope!

**Other information**

### CODE REVIEW

**Comment by TheLarkInn** [comment]
@sokra How should I fix these unit tests?! I totally forgot that we had some for these plugins.

File: lib/AmdMainTemplatePlugin.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ class AmdMainTemplatePlugin {
 	apply(compilation) {
 		const mainTemplate = compilation.mainTemplate;
 
-		compilation.templatesPlugin("render-with-entry", (source, chunk, hash) => {
+		compilation.tapMainAndChunkTemplates("AmdMainTemplatePlugin", "renderWithEntry", (source, chunk, hash) => {
```
Comment (sokra) [inline lib/AmdMainTemplatePlugin.js]: Use
``` js
for(const template of [compilation.mainTemplate, compilation.chunkTemplate]) {
  template.hooks.renderWithEntry(...);
}
```

Reason: I want the `.hooks.xxx` visible in source code to be able to search for `.hooks`.

Comment (sokra) [reply lib/AmdMainTemplatePlugin.js]: Alternative:

We add a `MultiHook` which works like this:

``` js
// Compilation contructor
this.bothTemplates = {
  hooks: {
    renderWithEntry: new MultiHook([
      compilation.mainTemplate.renderWithEntry, 
      compilation.chunkTemplate.renderWithEntry
    ])
  }
}
```

Comment (TheLarkInn) [reply lib/AmdMainTemplatePlugin.js]: I think I like your first suggestion because it allows us to keep the hooks in one place.

**Comment by ooflorent** [comment]
@TheLarkInn old unit tests are not compatible with hooks because they mock `plugin`. I think you should remove the test file.

**Comment by TheLarkInn** [comment]
I did @ooflorent thanks for the heads up.

**Pull request merged**